src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/OptimizedCompletionList.cs

```diff
@@ -0,0 +1,184 @@
+                    }
+                    writer.WriteEndArray();
+                }
+
+                writer.WriteEndObject();
+            }
+
+            private void WriteCompletionItem(JsonWriter writer, CompletionItem completionItem, JsonSerializer serializer)
+            {
+                writer.WriteStartObject();
+
+                var label = completionItem.Label;
+                if (label != null)
+                {
+                    writer.WritePropertyName("label");
+                    writer.WriteValue(label);
+                }
+
+                if (completionItem is VSLspCompletionItem lspCompletionItem && lspCompletionItem.Icon != null)
+                {
+                    writer.WritePropertyName("icon");
+                    if (lspCompletionItem.Icon == VSLspCompletionItemIcons.TagHelper)
+                    {
+                        writer.WriteRawValue(TagHelperIconRawJson);
+                    }
+                    else
+                    {
+                        serializer.Serialize(writer, lspCompletionItem.Icon);
+                    }
+                }
+
+                writer.WritePropertyName("kind");
+                writer.WriteValue(completionItem.Kind);
+
+                if (completionItem.Detail != null)
+                {
+                    writer.WritePropertyName("detail");
+                    writer.WriteValue(completionItem.Detail);
+                }
+
+                if (completionItem.Documentation != null)
+                {
+                    writer.WritePropertyName("documentation");
+                    serializer.Serialize(writer, completionItem.Documentation);
+                }
+
+                // Only render preselect if it's "true"
+                if (completionItem.Preselect)
+                {
+                    writer.WritePropertyName("preselect");
+                    writer.WriteValue(completionItem.Preselect);
+                }
+
+                if (completionItem.SortText != null && !string.Equals(completionItem.SortText, label, StringComparison.Ordinal))
+                {
+                    writer.WritePropertyName("sortText");
+                    writer.WriteValue(completionItem.SortText);
+                }
+
+                if (completionItem.FilterText != null && !string.Equals(completionItem.FilterText, label, StringComparison.Ordinal))
+                {
+                    writer.WritePropertyName("filterText");
+                    writer.WriteValue(completionItem.FilterText);
+                }
+
+                if (completionItem.InsertText != null && !string.Equals(completionItem.InsertText, label, StringComparison.Ordinal))
+                {
+                    writer.WritePropertyName("insertText");
+                    writer.WriteValue(completionItem.InsertText);
+                }
+
+                if (completionItem.InsertTextFormat != default)
```

Comment:
This is odd, isn't this an enum? Is this intended to test against PlainText?

Reply:
Ya but I choose to use "default" to be friendly to folks who aren't aware that PlainText is the default LSP spec option 😄

Reply:
But isn't default for an enum zero? As neither value for the enum is zero, will that optimization ever happen?

Reply:
Oh interesting, I had no idea that the `default` operator wasn't smart enough to take one of the known values of an enum (just tried it out). Really good call out! I'll fix